gitk

```diff
@@ -113,6 +113,33 @@ if {[is_Windows]} {
 
 # End of safe PATH lookup stuff
 
+proc have_pbcopy {} {
+    if {$::tcl_platform(os) ne {Darwin}} {
+        return 0
+    }
+    if {[catch { exec -ignorestderr -- pbcopy -help 2>@1 } helptext] != 0 } {
+        return 0
+    }
+    if {![string match "*Usage: pbcopy*" $helptext]} {
+        return 0
+    }
+    return 1
+}
+
+if {[have_pbcopy]} {
+    proc set_clipboard {text} {
+        # Using `pbcopy` works around
+        # https://core.tcl-lang.org/tk/tktview/e94c8bc845b4d5a658cd5421ef23ef2484560fa8
+        # `clipboard append` trims the last newline, so we do too.
```

Comment:
I'm confused about this. When I try `clipboard append "a\n"`, the `\n` goes to the clipboard. However, without `string trimright` below, the result of `gitk`'s copies to the clipboard with `pbcopy` have an extra trailing newline compared to the behavior with `clipboard append`. Perhaps the behavior of `clipboard append` is more confusing than I thought, intentionally or because of a bug.